src/app/packages/menu-items/filters/effects/filters.effects.ts

```diff
@@ -41,20 +42,28 @@ export class FiltersEffects {
     constructor(private actions$: Actions,
         private filtersService: FiltersService,
         private store: Store<IFiltersState>,
-        @Inject(FilterMetadata) private filtersMetadata: FilterMetadata[]) { }
+        private injector: Injector) { }
```

Comment:
I do not think that get the injector is good practice.
You can create the GenericTypeResolver as angular service.
If you want to get to the eadg. You  can create class that independent from the injector.
And Implement angular service as app util that use the injector.